src/server/mod.rs

```diff
@@ -672,23 +669,21 @@ impl LanguageServer for LspServer {
             Err(err) => return Err(err.into()),
         };
 
-        let mut visitor = semantic::FoldFinderVisitor::default();
-        let pkg_node = walk::Node::Package(&pkg);
-
-        walk::walk(&mut visitor, pkg_node);
-
-        let nodes = visitor.nodes;
-
-        let mut results = vec![];
-        for node in nodes {
-            results.push(lsp::FoldingRange {
+        let visitor = crate::walk_semantic_package!(
+            semantic::FoldFinderVisitor::default(),
+            pkg
+        );
+        let results: Vec<lsp::FoldingRange> = visitor
+            .nodes
+            .into_iter()
+            .map(|node| lsp::FoldingRange {
                 start_line: node.loc().start.line,
                 start_character: Some(node.loc().start.column),
                 end_line: node.loc().end.line,
                 end_character: Some(node.loc().end.column),
                 kind: Some(lsp::FoldingRangeKind::Region),
             })
-        }
+            .collect();
```

Comment:
Nice! This is cleaner! Just checking, is `.into_iter()` a visitor function to visit all the nodes?

Reply:
`into_iter` creates an iterator that consumes each value. The effect is that the ownership of the values changes to the iterator, and thus the new resulting `Vec<lsp::FoldingRange>`. If I only wanted to borrow the values (which would have been fine here, but I was trying to be explicit that we don't need the values anymore), I could just use `iter`, and the map function would take a borrowed value rather than taking full ownership of it.

Admittedly, this probably shouldn't have been part of this patch, but the change in how we reference `visitor` meant it was shallow to do. I have a followup that reduces `let mut` a bit more because of this change.